tsconfig.base.json

```diff
@@ -14,6 +14,8 @@
 		"emitDeclarationOnly": true,
 		"isolatedModules": true,
 
+		"skipDefaultLibCheck": true,
```

Comment:
TS doc recommends that we use `skipLibCheck` instead. Any good reason to prefer `skipDefaultLibCheck`?

Reply:
I noticed that, I'm not sure why they recommend that.

`skipLibCheck` ensures that the libraries we depend on typecheck. I understand that `skipDefaultLibCheck` only skips checking bundle libs (like DOM, ES2020, etc.). These libs are bundled with TypeScript and they should be the highest quality.

With other libs that come from other node_modules, it seems best to check their types especially because if we don't check these files, the errors can appear in other projects that have `skipLibCheck` disabled. We've seen that packages become stale or have conflicting types that we need to address, I dealt with a lot of this recently in https://github.com/WordPress/gutenberg/pull/61486 and related PRs.

Reply:
Makes sense to me. Seems like TS docs need some love!